common/src/main/java/com/wynntils/functions/generic/MathFunctions.java

```diff
@@ -0,0 +1,161 @@
+                    + arguments.getArgument("second").getDoubleValue();
+        }
+
+        @Override
+        public FunctionArguments.Builder getArgumentsBuilder() {
+            return new FunctionArguments.Builder(List.of(
+                    new FunctionArguments.Argument<>("first", Number.class, null),
+                    new FunctionArguments.Argument<>("second", Number.class, null)));
+        }
+    }
+
+    public static class SubtractFunction extends GenericFunction<Double> {
+        @Override
+        public Double getValue(FunctionArguments arguments) {
+            return arguments.getArgument("first").getDoubleValue()
+                    - arguments.getArgument("second").getDoubleValue();
+        }
+
+        @Override
+        public FunctionArguments.Builder getArgumentsBuilder() {
+            return new FunctionArguments.Builder(List.of(
+                    new FunctionArguments.Argument<>("first", Number.class, null),
+                    new FunctionArguments.Argument<>("second", Number.class, null)));
+        }
+
+        @Override
+        public List<String> getAliases() {
+            return List.of("sub");
+        }
+    }
+
+    public static class MultiplyFunction extends GenericFunction<Double> {
+        @Override
+        public Double getValue(FunctionArguments arguments) {
+            return arguments.getArgument("first").getDoubleValue()
+                    * arguments.getArgument("second").getDoubleValue();
+        }
+
+        @Override
+        public FunctionArguments.Builder getArgumentsBuilder() {
+            return new FunctionArguments.Builder(List.of(
+                    new FunctionArguments.Argument<>("first", Number.class, null),
+                    new FunctionArguments.Argument<>("second", Number.class, null)));
+        }
+
+        @Override
+        public List<String> getAliases() {
+            return List.of("mul");
+        }
+    }
+
+    public static class DivideFunction extends GenericFunction<Double> {
+        @Override
+        public Double getValue(FunctionArguments arguments) {
+            return arguments.getArgument("first").getDoubleValue()
+                    / arguments.getArgument("second").getDoubleValue();
+        }
+
+        @Override
+        public FunctionArguments.Builder getArgumentsBuilder() {
+            return new FunctionArguments.Builder(List.of(
+                    new FunctionArguments.Argument<>("first", Number.class, null),
+                    new FunctionArguments.Argument<>("second", Number.class, null)));
+        }
+
+        @Override
+        public List<String> getAliases() {
+            return List.of("div");
+        }
+    }
+
+    public static class ModuloFunction extends GenericFunction<Double> {
+        @Override
+        public Double getValue(FunctionArguments arguments) {
+            return arguments.getArgument("first").getDoubleValue()
+                    % arguments.getArgument("second").getDoubleValue();
```

Comment:
Kind of weird doing module a non-integer? Is that even well-defined?

Reply:
Fully defined within Java specs. Only C/C++ can't handle it. https://docs.oracle.com/javase/specs/jls/se17/html/jls-15.html#jls-15.17.3

Reply:
Goodness gracious... But, even if Java allows it, I think it makes no sense. I think we will help avoid problems for users if we limit the second argument to integers.